bin/create-ebs-volume

```diff
@@ -191,107 +182,92 @@ function create_and_attach_volume() {
+    # but this ends up making the '--encrypted' argument to fail during the execution of the install script. Conversely, NOT putting double-quotes
+    # around $tag_specification causes a parsing error due to the space in the $timestamp value (added to $tag_specification above).
+    local tmpfile=$(mktemp /tmp/ebs-autoscale.create-volume.XXXXXXXXXX)
+    local volume=$(\
+        aws ec2 create-volume \
+            --region $region \
+            --availability-zone $availability_zone \
+            $volume_opts \
+            --tag-specification "$tag_specification" \
+            2>> $tmpfile
+    )
+
+    if [ $? -ne 0 ]; then
+        logerr "Failed to create a volume, response was: $volume"
+        error "Error log: $(cat $tmpfile)"
+    fi
 
+    local volume_id=`echo $volume | jq -r '.VolumeId'`
     if [ -z "$volume_id" ]; then
-        logthis "$(cat $tmpfile)"  # log captured error
-        cat $tmpfile               # print captured error (e.g. when called during install)
-        rm $tmpfile
-
-        error "could not create volume"
+        logerr "Failed to parse volume id from response: $volume"
+        error "Error log: $(cat $tmpfile | tr '\n' ,)"
     fi
-    rm $tmpfile
 
-    logthis "created volume: $volume_id [ $volume_opts ]"
+    loginfo "created volume: $volume_id [ $volume_opts ]"
 
     # In theory this shouldn't need to loop as aws ec2 wait will retry but I have seen it exceed request limits
-    for i in {1..3} ; do
-       if aws ec2 wait volume-available --region $region --volume-ids $volume_id; then
-         logthis "volume $volume_id available"
-         break
-       fi
-    done
+    aws ec2 wait volume-available \
+        --region $region \
+        --volume-ids $volume_id \
+        2>> $tmpfile 1>&2
+
+    if [ $? -ne 0 ] ; then
+        aws ec2 delete-volume --region $region --volume-id $volume_id 2>> $tmpfile
+        logerr "Failed to wait for volume available"
+        error "Error log: $(cat $tmpfile | tr '\n' ,)"
+    fi
 
     # Need to assure that the created volume is successfully attached to be
     # cost efficient.  If attachment fails, delete the volume.
-    set +e
-    logthis "attaching volume $volume_id"
+    loginfo "attaching volume $volume_id"
 
     sleep 1
     aws ec2 attach-volume \
         --region $region \
         --device $device \
         --instance-id $instance_id \
         --volume-id $volume_id \
-    > /dev/null
+        2>> $tmpfile 1>&2
     
     status="$?"
     if [ ! "$status" -eq 0 ]; then
-        logthis "deleting volume $volume_id"
-        aws ec2 delete-volume \
-            --region $region \
-            --volume-id $volume_id \
-        > /dev/null
-        
-        error "could not attach volume to instance"
+        logerr "deleting volume $volume_id"
+        aws ec2 delete-volume --region $region --volume-id $volume_id 2>> $tmpfile
```

Comment:
maybe arguments in the next line like other aws commands in the file to be consistent ??